pkg/auth/authHost.go

```diff
@@ -0,0 +1,85 @@
+package auth
+
+import (
+	"fmt"
+	"math/rand"
+	"net/url"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/snyk/go-application-framework/internal/api"
+)
+
+func redirectAuthHost(instance string) (string, error) {
+	// handle both cases if instance is a URL or just a host
+	if !strings.HasPrefix(instance, "http") {
+		instance = "https://" + instance
+	}
+
+	instanceUrl, err := url.Parse(instance)
+	if err != nil {
+		return "", err
+	}
+
+	canonicalizedInstanceUrl, err := api.GetCanonicalApiAsUrl(*instanceUrl)
+	if err != nil {
+		return "", err
+	}
+
+	return canonicalizedInstanceUrl.Host, nil
+}
+
+func isValidAuthHost(authHost string, hostRegularExpression string) (bool, error) {
+	if len(hostRegularExpression) == 0 {
+		return false, fmt.Errorf("regular expression to check host names must not be empty")
+	}
+
+	r, err := regexp.Compile(hostRegularExpression)
+	if err != nil {
+		return false, err
+	}
+
+	return r.MatchString(authHost), nil
+}
+
+// FilterSupportedPatRegions iterates a list of region URLs and filters out regions not supported by PAT.
+// It returns a filtered slice of URL strings
+func FilterSupportedPatRegions(regions []string, unsupportedRegions []string) []string {
```

Comment:
Suggestion: Please write some Unit tests. I assume that this logic can be simplified but without unit tests it is difficult to fully follow what the intention is.